apps/rush-lib/src/cli/actions/BaseRushAction.ts

```diff
@@ -11,19 +12,43 @@ import {
 import RushConfiguration from '../../data/RushConfiguration';
 import EventHooksManager from '../logic/EventHooksManager';
 
+export interface IRushCommandLineActionOptions extends ICommandLineActionOptions {
+  /** If true, no locking mechanism will be enforced when this action is run */
+  safeForSimultaneousRushProcesses?: boolean;
+}
+
 /**
  * The base Rush action that all Rush actions should extend.
  */
 export abstract class BaseRushAction extends CommandLineAction {
   private _rushConfiguration: RushConfiguration;
   private _eventHooksManager: EventHooksManager;
+  private _safeForSimultaneousRushProcesses: boolean;
 
-  constructor(options: ICommandLineActionOptions) {
+  constructor(options: IRushCommandLineActionOptions) {
     super(options);
+    this._safeForSimultaneousRushProcesses = !!options.safeForSimultaneousRushProcesses;
   }
 
   protected onExecute(): void {
     this._ensureEnvironment();
+
+    if (!this._safeForSimultaneousRushProcesses) {
+      const lockFilePath: string = path.join(
+        this.rushConfiguration.commonTempFolder,
+        'rush.lock'
+      );
+      if (fsx.existsSync(lockFilePath)) {
+        fsx.removeSync(lockFilePath);
+      }
```

Comment:
You're always deleting the lock file? That doesn't seem right.

Reply:
I'll add a comment about this, but it boils down to weirdness in NodeJS.

Reply:
This weird code is a workaround we came up with because NodeJS neglects to provide access to the real OS APIs for file locking.  It definitely needs some comments to explain what's going on.

---
In reply to: [165489834](https://github.com/Microsoft/web-build-tools/pull/517#discussion_r165489834) [](ancestors = 165489834)